test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env rspec
+
+root_path = File.expand_path('../..', __FILE__)
+ENV["Y2DIR"] = File.join(root_path, 'src')
+
+require "yast"
+require_relative "../../../src/include/autoinstall/ask.rb"
+require_relative "../../../src/modules/Profile"
+
+describe "Yast::AutoinstallAskInclude" do
+  module DummyYast
+    class AutoinstallAskClient < Yast::Client
+      def main
+        Yast.include self, "autoinstall/ask.rb"
+      end
+
+      def initialize
+        main
+      end
+    end
+  end
+
+  let(:client) { DummyYast::AutoinstallAskClient.new }
+  let(:base_ask) do
+    { "type" => "string", "question" => "hostname?", "default" => "my.site.de",
+      "help" => "Some help", "dialog" => 0, "element" => 0,
+      "stage" => "initial" }
+  end
+  let(:profile) { { "general" => { "ask-list" => ask_list } } }
+  let(:result) { :ok }
+
+  describe "#askDialog" do
+    let(:ask_list) { [ask] }
+
+    before(:each) do
+      allow(Yast::Profile).to receive(:current).and_return(profile)
+      allow(Yast::Stage).to receive(:initial).and_return(true)
+      allow(Yast::UI).to receive(:UserInput).and_return(result)
+    end
+
+    context "when ask-list is empty" do
+      let(:ask_list) { [] }
+
+      it "no dialog is shown" do
+        expect(Yast::UI).to_not receive(:OpenDialog)
+        client.askDialog
+      end
+    end
+
+    describe "dialogs creation" do
+      context "when the ask-list contains a question with type 'string'" do
+        let(:ask) { base_ask }
+
+        it "creates a TextEntry widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expect(client).to receive(:TextEntry).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], ask["default"]).
+            and_call_original
```

Comment:
btw this is not nice way to test UI. I think better way is to do introspection of dialog in openDialog. At least it is my intention when adding introspection to term.
https://github.com/yast/yast-ruby-bindings/blob/master/src/ruby/yast/term.rb#L69
I think we can create some helper for rspec for openDialog to check if there is given element...for now you can use something like ( just pseudo code!!!)
```
expect(Yast::UI).to receive(:OpenDialog) do |dialog, options|
  text_entry = dialog.nested_find do |w|
     w.value == :TextEntry && w.include?(Id("0_0"))
  end

  expect(text_entry).to include Opt(:notify)
  expect(text_entry).to include ask["question"]
  expect(text_entry).to include ask["default"]
end